docs/DESIGN.md

```diff
@@ -104,7 +104,6 @@ The IPXE binaries are built from source and then embedded into the Boots Go bina
 
 ### Building the IPXE binary
 
-The scripts in this repo to build the IPXE binaries only work on Linux.
```

Comment:
Is this accurate? I think the mac issue is really that at this nixpkgs pin the gcc cross compiler is not pre-built, otherwise it should work for mac (or would be easy to make work for mac). Would be good to see if its prebuilt in hydra in a different revision and/or ask in matrix about enabling for mac (if not pre-built on purpose).

Reply:
Regardless of the gcc cross compiler package concerns, I am not able to build any IPXE binaries on my Mac. I've tried unsuccessfully to get it working. Below is just the latest error I'm running into.
```bash
nix-shell
make bindata
.
.
.
  [BUILD] bin-x86_64-efi/__divdi3.o
In file included from <command-line>:
./include/compiler.h:219: error: "__weak" redefined [-Werror]
  219 | #define __weak  __attribute__ (( weak, noinline ))
      | 
<built-in>: note: this is the location of the previous definition
cc1: all warnings being treated as errors
make[1]: *** [Makefile.housekeeping:943: bin-x86_64-efi/__divdi3.o] Error 1
make[1]: Leaving directory '/tinkerbell/boots/ipxe/ipxe/build/bin-x86_64-efi/ipxe-a5fb41873dc1dcce5dcc817bf53a0649c01f8cac/src'
make: *** [rules.mk:81: ipxe/ipxe/build/bin-x86_64-efi/ipxe.efi] Error 2
```

Reply:
@jacobweinstock want to try bumping ipxe to the latest git rev?

Reply:
@mmlb sure. how do I do that?

Reply:
Update the values in https://github.com/tinkerbell/boots/blob/master/ipxev.mk, commit and the sha512sum of the archive tarball you can fetch from github.

Reply:
@mmlb same error

Reply:
ok cool, we *could* also try a new nixpkgs (in case it has newer gcc that fixes this too)

Reply:
@mmlb , tried that too. same error.